File: specification/specs/x_pack/info/x_pack_usage/XPackUsageResponse.ts
/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

class XPackUsageResponse extends ResponseBase {
  watcher: AlertingUsage
  ccr: CcrUsage
  data_frame: XPackUsage
  data_science: XPackUsage
  enrich: XPackUsage
  flattened: FlattenedUsage
  graph: XPackUsage
  ilm: IlmUsage
  logstash: XPackUsage
  ml: MachineLearningUsage
  monitoring: MonitoringUsage
  rollup: XPackUsage
  security: SecurityUsage
  slm: SlmUsage
  sql: SqlUsage
  transform: XPackUsage
  vectors: VectorUsage
  voting_only: XPackUsage
}

File: output/csharp/XPack/Info/XPackUsage/XPackUsageResponse.cs
using Nest.XPack;
using Nest.CommonAbstractions;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.XPack {

	public class XPackUsageResponse : IResponse {
		
		[DataMember(Name="watcher")]
		public AlertingUsage Watcher { get; set; }


		[DataMember(Name="ccr")]
		public CcrUsage Ccr { get; set; }


		[DataMember(Name="data_frame")]
		public XPackUsage DataFrame { get; set; }


		[DataMember(Name="data_science")]
		public XPackUsage DataScience { get; set; }


		[DataMember(Name="enrich")]
		public XPackUsage Enrich { get; set; }


		[DataMember(Name="flattened")]
		public FlattenedUsage Flattened { get; set; }


		[DataMember(Name="graph")]
		public XPackUsage Graph { get; set; }


		[DataMember(Name="ilm")]
		public IlmUsage Ilm { get; set; }


		[DataMember(Name="logstash")]
		public XPackUsage Logstash { get; set; }


		[DataMember(Name="ml")]
		public MachineLearningUsage Ml { get; set; }


		[DataMember(Name="monitoring")]
		public MonitoringUsage Monitoring { get; set; }


		[DataMember(Name="rollup")]
		public XPackUsage Rollup { get; set; }


		[DataMember(Name="security")]
		public SecurityUsage Security { get; set; }


		[DataMember(Name="slm")]
		public SlmUsage Slm { get; set; }


		[DataMember(Name="sql")]
		public SqlUsage Sql { get; set; }


		[DataMember(Name="transform")]
		public XPackUsage Transform { get; set; }


		[DataMember(Name="vectors")]
		public VectorUsage Vectors { get; set; }


		[DataMember(Name="voting_only")]
		public XPackUsage VotingOnly { get; set; }

	}
}
